src/command/marian_server.cpp

```diff
@@ -14,22 +14,29 @@ int main(int argc, char **argv) {
   // Initialize translation task
   auto options = parseOptions(argc, argv, cli::mode::server, true);
   auto task = New<TranslateService<BeamSearch>>(options);
+  auto inputNum = options->get<std::vector<std::string>>("vocabs").size() - 1;
 
   // Initialize web server
   WSServer server;
   server.config.port = (short)options->get<size_t>("port", 8080);
 
   auto &translate = server.endpoint["^/translate/?$"];
 
-  translate.on_message = [&task](Ptr<WSServer::Connection> connection,
+  translate.on_message = [&task, inputNum](Ptr<WSServer::Connection> connection,
                                  Ptr<WSServer::Message> message) {
     // Get input text
     auto inputText = message->string();
     auto sendStream = std::make_shared<WSServer::SendStream>();
 
     // Translate
     timer::Timer timer;
-    auto outputText = task->run(inputText);
+    std::vector<std::string> inputs;
+    if (inputNum >= 2) {
+      inputs = utils::tsv2lists(inputText, inputNum);
+    } else {
+      inputs = std::vector<std::string>({inputText});
+    }
+    auto outputText = task->run(inputs);
```

Comment:
With this logic `std::string run(std::string)` is used nowhere, am I right? So either this version of the function can be removed, or the logic here can be something similar to this:
```
std::string o;
if(inputNum >= 2) {
  o = task->run(utils::tsv2lists(inputText, inputNum));
} else {
  o = task->run(inputText);
}
```
I think the latter is cleaner, but feel free to object.